### PR TITLE
Add data source for github_actions_environment_public_key

### LINKED ISSUE #1769: [FEAT]: DataSource for github_actions_environment_public_key
### Describe the need

When setting secrets for an action environment, you need to use the key for that environment. There is currently no datasource for fetching this particular key.

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ijcd**
Resolves #1769

----

### Before the change?

* When setting secrets for an action environment, you need to use the key for that environment. There is currently no datasource for fetching this particular key.

 

### After the change?

* There is now a data source for github_actions_environment_public_key

### Other information

* 

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

I'm looking for guidance on how to best test this.
I'm also looking for guidance on how to update the docs. Are they auto-generated or does one edit stuff under `website/`?

### Does this introduce a breaking change?

No

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Comment by kfcampbell** [comment]
@ijcd would you mind creating a documentation page for this new data source and linking to it in `github.erb`?

**Comment by ijcd** [comment]
I believe I have added the docs correctly. Please take a look. I can't figure out how to build the website to test that, however. Any pointers?

```
% make website
make[1]: *** No rule to make target `website-provider'.  Stop.
make: *** [website] Error 2
```

File: github/data_source_github_actions_environment_public_key_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccGithubActionsEnvironmentPublicKeyDataSource(t *testing.T) {
+
+	t.Run("queries an environment public key without error", func(t *testing.T) {
+
+		config := `
```
Comment (kfcampbell) [inline github/data_source_github_actions_environment_public_key_test.go:13]: Currently, this test will not run as the configuration is missing the required environment and repository parameters. You can look at `github/data_source_github_actions_environment_secrets_test.go` for an example of how to construct a repo and environment in a test case.

**Comment by kfcampbell** [comment]
@ijcd Thanks for your patience! The docs you've added look good. The make commands concerning the website are unfortunately not working (I think since the handoff from Hashicorp? I'm not positive). I've added one comment about testing this functionality and we should be good to go here once tests are all set.

**Comment by ijcd** [comment]
I added to the test case. I had to do this blind because I can't figure out how to get the tests to run. They either complain about things like `skipUnlessMode` if I try to run them one at a time, or that they are missing lots of env vars if I try to run the suite. Doing what is in `ci.yml` isn't working either. I'm missing these:

```
GITHUB_TEMPLATE_REPOSITORY_RELEASE_ID
GITHUB_TEMPLATE_REPOSITORY
GITHUB_TEST_COLLABORATOR
GITHUB_TEST_USER
GITHUB_TOKEN
GITHUB_ORGANIZATION
```

**Pull request closed without merging**